sympy/parsing/sympy_parser.py

```diff
@@ -1086,13 +1089,36 @@ def parse_expr(s, local_dict=None, transformations=standard_transformations,
     global_dict['max'] = Max
     global_dict['min'] = Min
 
+    # Apply transformations. Result is still a string.
     code = stringify_expr(s, local_dict, global_dict, transformations)
 
+    # Now we move to an AST
+    node = ast.parse(code)
     if not evaluate:
-        code = compile(evaluateFalse(code), '<string>', 'eval')
+        node = EvaluateFalseTransformer().visit(node)
+        node = ast.fix_missing_locations(node)
+    expr_value = node.body[0].value
 
     try:
-        rv = eval_expr(code, local_dict, global_dict)
+        if controlled:
+            # Here evaluation is achieved by applying another node visitor
+            # to the AST.
+            ce = ControlledEvaluator(local_dict, global_dict)
+            node = ast.Expr(expr_value)
+            rv = ce.visit(node)
+        else:
+            # FIXME: the assertion was just here to ensure we were testing the
+            #  new method in all unit tests. What's the right way to test?
+            #  Duplicate all tests in `test_sympify.py` and `test_sympy_parser.py`?
+            assert False
```

Comment:
You can also trigger this with `sympify("parse_expr('__import__(\"os\").system(\"ls\")', controlled=False)")`.

Reply:
Ah, so `parse_expr` would have to be removed from the whitelist as well (like `sympify`).